src/main/resources/errors.yaml

```diff
@@ -273,6 +273,18 @@ request-errors:
       
       Note, DeleteMany without a filter truncates the table.
 
+  - scope: FILTER
+    code: INVALID_IN_FILTER_FOR_UPDATE_ONE_DELETE_ONE
+    title: In filter is not allowed for UpdateOne and DeleteOne commands
+    body: |-
+      The command used a $in or $nin filter to select the rows the modify, however these filters can select more than one row.
```

Comment:
I'm sure there's an internal reason to disallow calling `updateOne()` with `$in`/`$nin`, but this sentence doesn't seem to justify why you shouldn't use `$in` or `$nin` with `updateOne()`.

For example, a common pattern I've seen with `updateOne()` is `updateOne({ _id: 42, status: { $in: ['pending', 'active'] } }, update)`: update the document with `_id: 42` if it has status 'pending' or 'active'. This pattern can only update at most one row.

Similarly, `updateOne({ isDeleted: false })` can select many rows.

Not critical, but it would be nice to make this error message more accurately summarize the reasoning for this behavior.